mlflow/utils/environment.py

```diff
@@ -748,7 +748,7 @@ def _get_mlflow_env_name(s):
         (e.g. "mlflow-da39a3ee5e6b4b0d3255bfef95601890afd80709")
 
     """
-    return "mlflow-" + hashlib.sha1(s.encode("utf-8"), usedforsecurity=False).hexdigest()
+    return "mlflow-" + hashlib.sha256(s.encode("utf-8")).hexdigest()
```

Comment:
What's going to happen if we ship this?

Let's say you have a model and the environment created by mlflow for that model. If we ship this and you install mlflow 2.19.0, mlflow will create a new environment because the environment ID is not the same.

Reply:
One concern:

Now we have "prebuilt" environment functionality.
The change will cause python env built by old MLflow version can't be used by new mlflow version.

But the functionality is very new so the impact should be small.

Reply:
Yes, the impact is low.